### PR TITLE
Switch documentation Template and add typescript support to docs

### PR DESCRIPTION
**Pull request by arthuro555**
As discussed in #1669

### CODE REVIEW

File: GDJS/docs/jsdoc.conf.json [hunk 1]
```diff
@@ -23,6 +23,8 @@
       "outputSourceFiles": true,
       "includeDate": false
```
Comment (4ian) [inline GDJS/docs/jsdoc.conf.json:24]: Can you verify if these options are still necessary?

**Comment by arthuro555** [comment]
image  
That's how braintree looks like.  
image  
And that's better-docs.

I personally prefer better docs because
1. It supports interfaces
2. It supports typescript
3. It supports searching
4. It has that nice navigation toolbar on the right to navigate through a class

And IMO it looks as clean as the other.

File: GDJS/Runtime/gd.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ gdjs.toDegrees = function (angleInRadians) {
  * of the type of the object is "TextObject::Text".
```
Comment (4ian) [inline GDJS/Runtime/gd.js:135]: These changes are not right, the parameters are NOT `gdjs.Runtimeobject`, which would expect an instance of the class, but the class itself.
If it's causing problems, try `Class<gdjs.RuntimeObject>`?

**Comment by 4ian** [comment]
I like better-docs, looks clean. We can go ahead with this.

**Comment by 4ian** [comment]
Typescript was failing, I fixed this for you. Looks good to go, we can always switch to another theme later as links are not changed.
While browsing the docs, I think we should:
- Add a link in the main page to the Type Definitions, that are hidden in global.
- Hide the Cocos related classes.
- Maybe start to index also the extension, at least the official ones so that people writing JavaScript code can see the documentation of the gdjs.TiledSpriteRuntimeObject, etc...
- Add more `// @ts-check` and fix our types/docs. There are still a lot of bad parameters, missing types, etc! Would be good to make this more professional by completing/fixing stuff progressively.

**Pull request merged**